crates/opencascade-sys/build.rs

```diff
@@ -1,10 +1,94 @@
+use std::path::PathBuf;
+
+struct OccVersion {
+    major: u8,
+    minor: u8,
+}
+
+struct OccConfig {
+    include_dir: PathBuf,
+    library_dir: PathBuf,
+    is_dynamic: bool,
+}
+
+const OCC_VERSION: OccVersion = OccVersion { major: 7, minor: 2 };
+
+const OCC_LIBS: &[&str] = &[
+    "TKMath",
+    "TKernel",
+    "TKFeat",
+    "TKGeomBase",
+    "TKG2d",
+    "TKG3d",
+    "TKTopAlgo",
+    "TKGeomAlgo",
+    "TKGeomBase",
+    "TKBRep",
+    "TKPrim",
+    "TKSTEP",
+    "TKSTEPAttr",
+    "TKSTEPBase",
+    "TKSTEP209",
+    "TKSTL",
+    "TKMesh",
+    "TKShHealing",
+    "TKFillet",
+    "TKBool",
+    "TKBO",
+    "TKOffset",
+    "TKV3d",
+    "TKXSBase",
+];
+
 fn main() {
     let target = std::env::var("TARGET").expect("No TARGET environment variable defined");
     let is_windows = target.to_lowercase().contains("windows");
     let is_windows_gnu = target.to_lowercase().contains("windows-gnu");
+    // prefer dynamic linking
+    let mut is_dynamic = cfg!(feature = "dynamic");
+    // prefer builtin OCC
+    let is_builtin = cfg!(feature = "builtin");
+
+    if is_windows {
+        println!("cargo:rustc-link-lib=dylib=user32");
+    }
+
+    let mut build = cxx_build::bridge("src/lib.rs");
+
+    if is_windows_gnu {
+        build.define("OCC_CONVERT_SIGNALS", "TRUE");
+    }
+
+    // Try find packaged version of OpenCASCADE or use builtin
+    let occ_config = if is_builtin { None } else { find_opencascade(is_dynamic, OCC_VERSION) }
+        .unwrap_or_else(|| build_opencascade(is_dynamic));
```

Comment:
This falls back to the bundled OpenCascade if `dynamic` is set but we cannot find it. That feels a bit too magic/surprising - I think it would be better to fail in this case, tell the user to try changing cargo flags instead.

Reply:
I tried preserve current behavior as much as possible so I think the falling back to builtin version is a right thing.

Reply:
> I tried preserve current behavior as much as possible

Oh yeah that's a good point, we don't want to break the build for people, and using the built-in OCC is more universal.

However I think that should be solved by making the `builtin` feature flag enabled by default on `opencascade` level.

The automatic fallback could be very confusing: imagine people disabling the `builtin` flag and still getting horrible build times (or opencascade-rs not picking up their custom-patched system-installed OCC) - perhaps they've installed OCC system-wide, but they version is not compatible. They would not be able to know the fact because the build won't tell them (output of build scripts is only visible with `cargo ... -vv`).